src/execution/operator/order/physical_order.cpp

```diff
@@ -1,105 +1,882 @@
-	// finalize: perform the actual sorting
-	auto &sink = (OrderByGlobalOperatorState &)*state;
-	ChunkCollection &big_data = sink.sorted_data;
+void PhysicalOrder::Combine(ExecutionContext &context, GlobalOperatorState &gstate_p, LocalSinkState &lstate_p) {
+	auto &gstate = (OrderGlobalState &)gstate_p;
+	auto &lstate = (OrderLocalState &)lstate_p;
 
-	// compute the sorting columns from the input data
-	ExpressionExecutor executor;
-	vector<LogicalType> sort_types;
-	vector<OrderType> order_types;
-	vector<OrderByNullType> null_order_types;
-	for (idx_t i = 0; i < orders.size(); i++) {
-		auto &expr = orders[i].expression;
-		sort_types.push_back(expr->return_type);
-		order_types.push_back(orders[i].type);
-		null_order_types.push_back(orders[i].null_order);
-		executor.AddExpression(*expr);
+	if (!lstate.sorting_block) {
+		return;
+	}
+
+	lock_guard<mutex> append_lock(gstate.lock);
+	for (auto &block : lstate.sorting_block->blocks) {
+		gstate.sorting_block->count += block.count;
+		gstate.sorting_block->blocks.push_back(move(block));
+	}
+	for (idx_t i = 0; i < lstate.var_sorting_blocks.size(); i++) {
+		if (!lstate.var_sorting_blocks[i]) {
+			continue;
+		}
+		for (auto &block : lstate.var_sorting_blocks[i]->blocks) {
+			gstate.var_sorting_blocks[i]->count += block.count;
+			gstate.var_sorting_blocks[i]->blocks.push_back(move(block));
+		}
+		for (auto &block : lstate.var_sorting_sizes[i]->blocks) {
+			gstate.var_sorting_sizes[i]->count += block.count;
+			gstate.var_sorting_sizes[i]->blocks.push_back(move(block));
+		}
+	}
+	for (auto &block : lstate.payload_block->blocks) {
+		gstate.payload_block->count += block.count;
+		gstate.payload_block->blocks.push_back(move(block));
+	}
+
+	const auto &payload_state = *gstate.payload_state;
+	if (payload_state.HAS_VARIABLE_SIZE) {
+		for (auto &block : lstate.sizes_block->blocks) {
+			gstate.sizes_block->count += block.count;
+			gstate.sizes_block->blocks.push_back(move(block));
+		}
+	}
+}
+
+static void RadixSort(BufferManager &buffer_manager, data_ptr_t dataptr, const idx_t &count, const idx_t &col_offset,
+                      const idx_t &sorting_size, const SortingState &sorting_state) {
+	auto temp_block =
+	    buffer_manager.Allocate(MaxValue(count * sorting_state.ENTRY_SIZE, (idx_t)Storage::BLOCK_ALLOC_SIZE));
+	data_ptr_t temp = temp_block->node->buffer;
+	bool swap = false;
+
+	idx_t counts[256];
+	uint8_t byte;
+	for (idx_t offset = col_offset + sorting_size - 1; offset + 1 > col_offset; offset--) {
+		// init to 0
+		memset(counts, 0, sizeof(counts));
+		// collect counts
+		for (idx_t i = 0; i < count; i++) {
+			byte = *(dataptr + i * sorting_state.ENTRY_SIZE + offset);
+			counts[byte]++;
+		}
+		// compute offsets from counts
+		for (idx_t val = 1; val < 256; val++) {
+			counts[val] = counts[val] + counts[val - 1];
+		}
+		// re-order the data in temporary array
+		for (int i = count - 1; i >= 0; i--) {
```

Comment:
Perhaps better to replace the int here with an idx_t, or at least int32_t?

In order to use an idx_t for a reverse loop like this I generally do:
```
for(idx_t i = count; i > 0; i--) {
   auto idx = i - 1;
   ...
}
```